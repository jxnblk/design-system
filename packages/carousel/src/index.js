export * from './Carousel'
export * from './ArrowButton'
