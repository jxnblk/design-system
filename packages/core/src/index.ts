import * as storybookArgs from './storybook/args'

export type { IButtonProps } from './Button'
export type { IDialogProps } from './Dialog'
export type { IGridProps } from './Grid'
export type { ICloseButtonProps } from './CloseButton'

export { Absolute } from './Absolute'
export { Accordion } from './Accordion'
export * from './Animate'
export { Avatar } from './Avatar'
export { BackgroundImage } from './BackgroundImage'
export { Badge } from './Badge'
export { Banner } from './Banner'
export { BlockLink } from './BlockLink'
export { Box } from './Box'
export { Breadcrumbs } from './Breadcrumbs'
export { Button } from './Button'
export { Card } from './Card'
export { Checkbox } from './Checkbox'
export { ButtonChip, FilterChip as Chip, ChoiceChip, FilterChip } from './Chip'
export { CloseButton } from './CloseButton'
export * from './Container'
export { Dialog } from './Dialog'
export { Divider } from './Divider'
export { DotLoader } from './DotLoader'
export { Flag } from './Flag'
export { Flex } from './Flex'
export { FormField, InputField } from './FormField'
export { GenericBanner } from './GenericBanner'
export * from './Grid'
export { Heading } from './Heading'
export { Hide } from './Hide'
export { Hug } from './Hug'
export { Icon } from './Icon'
export { IconButton } from './IconButton'
export { IconField } from './IconField'
export { Image } from './Image'
export { Input } from './Input'
export { InputGroup } from './InputGroup'
export { Label } from './Label'
export * from './Layout'
export { Link } from './Link'
export { List } from './List'
export { PasswordInput } from './PasswordInput'
export { PlaceholderImage } from './PlaceholderImage'
export * from './Popout'
export * from './ProgressBar'
export { Radio } from './Radio'
export { RadioCheckToggleCard } from './RadioCheckToggleCard'
export { RatingBadge } from './RatingBadge'
export { Relative } from './Relative'
export { Select } from './Select'
export { ShadowEffect, ShadowOverlay } from './ShadowEffect'
export { Shimmer } from './Shimmer'
export { SkipMenu } from './SkipMenu'
export { SlideBox } from './SlideBox'
export { Spinner } from './Spinner'
export { SrOnly } from './SrOnly'
export { Stamp } from './Stamp'
export { Stepper } from './Stepper'
export { Swatch } from './Swatch'
export { Text } from './Text'
export { TextArea } from './TextArea'
export { Toast } from './Toast'
export { ToastProvider } from './ToastProvider'
export { Toggle } from './Toggle'
export { ToggleBadge } from './ToggleBadge'
export { Tooltip } from './Tooltip'
export { Truncate } from './Truncate'

export { ThemeProvider } from './ThemeProvider'

export { useToast } from './ToastProvider'

export * from './utils'

// DocsUtils
export * from './DocsUtils/DoDont'
export * from './DocsUtils/DocTable'
export * from './DocsUtils/Hero'
// export * from './DocsUtils/LiveDemo'
export * from './DocsUtils/Note'
export * from './DocsUtils/RelatedComponent'
export * from './DocsUtils/Section'
export * from './DocsUtils/StoryHeading'
export * from './DocsUtils/StoryStage'
export * from './DocsUtils/TableOfContents'
export { storybookArgs }

export * from './theme'
