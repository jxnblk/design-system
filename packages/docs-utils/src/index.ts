/* istanbul ignore file */
export * from './components'
export * as utils from './utils'
