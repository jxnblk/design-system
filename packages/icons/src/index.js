export { default as Ac } from './Ac.jsx'
export { default as Accessible } from './Accessible.jsx'
export { default as AddAlarm } from './AddAlarm.jsx'
export { default as Airplane } from './Airplane.jsx'
export { default as Arrival } from './Arrival.jsx'
export { default as ArrowDown } from './ArrowDown.jsx'
export { default as ArrowLeft } from './ArrowLeft.jsx'
export { default as ArrowRight } from './ArrowRight.jsx'
export { default as ArrowUp } from './ArrowUp.jsx'
export { default as Attention } from './Attention.jsx'
export { default as Automatic } from './Automatic.jsx'
export { default as Backpack } from './Backpack.jsx'
export { default as Bag } from './Bag.jsx'
export { default as Beach } from './Beach.jsx'
export { default as Bed } from './Bed.jsx'
export { default as BedKing } from './BedKing.jsx'
export { default as BedSingle } from './BedSingle.jsx'
export { default as BeenHere } from './BeenHere.jsx'
export { default as Bike } from './Bike.jsx'
export { default as Bookmark } from './Bookmark.jsx'
export { default as BookmarkActivated } from './BookmarkActivated.jsx'
export { default as BoxChecked } from './BoxChecked.jsx'
export { default as BoxEmpty } from './BoxEmpty.jsx'
export { default as BoxMinus } from './BoxMinus.jsx'
export { default as BoxPlus } from './BoxPlus.jsx'
export { default as Breakfast } from './Breakfast.jsx'
export { default as Broom } from './Broom.jsx'
export { default as Build } from './Build.jsx'
export { default as Bus } from './Bus.jsx'
export { default as Business } from './Business.jsx'
export { default as Cake } from './Cake.jsx'
export { default as Calendar } from './Calendar.jsx'
export { default as CalendarCheck } from './CalendarCheck.jsx'
export { default as CalendarClock } from './CalendarClock.jsx'
export { default as CarAutomatic } from './CarAutomatic.jsx'
export { default as CarCircle } from './CarCircle.jsx'
export { default as CarDoor } from './CarDoor.jsx'
export { default as CarManual } from './CarManual.jsx'
export { default as CarWash } from './CarWash.jsx'
export { default as CardTravel } from './CardTravel.jsx'
export { default as Carriage } from './Carriage.jsx'
export { default as CarryOnBag } from './CarryOnBag.jsx'
export { default as Cars } from './Cars.jsx'
export { default as Casino } from './Casino.jsx'
export { default as ChargeStation } from './ChargeStation.jsx'
export { default as Chart } from './Chart.jsx'
export { default as ChartAlt } from './ChartAlt.jsx'
export { default as Chat } from './Chat.jsx'
export { default as ChatBubble } from './ChatBubble.jsx'
export { default as Check } from './Check.jsx'
export { default as CheckedBag } from './CheckedBag.jsx'
export { default as Checklist } from './Checklist.jsx'
export { default as ChevronDown } from './ChevronDown.jsx'
export { default as ChevronLeft } from './ChevronLeft.jsx'
export { default as ChevronRight } from './ChevronRight.jsx'
export { default as ChevronUp } from './ChevronUp.jsx'
export { default as CityView } from './CityView.jsx'
export { default as Clock } from './Clock.jsx'
export { default as Close } from './Close.jsx'
export { default as Cloud } from './Cloud.jsx'
export { default as CollisionCoverage } from './CollisionCoverage.jsx'
export { default as ContactMail } from './ContactMail.jsx'
export { default as ContactPhone } from './ContactPhone.jsx'
export { default as ConvenienceStore } from './ConvenienceStore.jsx'
export { default as Copy } from './Copy.jsx'
export { default as Couch } from './Couch.jsx'
export { default as Coupon } from './Coupon.jsx'
export { default as CreditCard } from './CreditCard.jsx'
export { default as Cruises } from './Cruises.jsx'
export { default as Departure } from './Departure.jsx'
export { default as Devices } from './Devices.jsx'
export { default as DevicesImportant } from './DevicesImportant.jsx'
export { default as Directions } from './Directions.jsx'
export { default as Discount } from './Discount.jsx'
export { default as Document } from './Document.jsx'
export { default as Dollar } from './Dollar.jsx'
export { default as DollarCircle } from './DollarCircle.jsx'
export { default as Dot } from './Dot.jsx'
export { default as DoubleBed } from './DoubleBed.jsx'
export { default as DoubleOccupancy } from './DoubleOccupancy.jsx'
export { default as Download } from './Download.jsx'
export { default as DownloadAlt } from './DownloadAlt.jsx'
export { default as EarlyBird } from './EarlyBird.jsx'
export { default as Edit } from './Edit.jsx'
export { default as Electric } from './Electric.jsx'
export { default as Email } from './Email.jsx'
export { default as Emoticon } from './Emoticon.jsx'
export { default as Event } from './Event.jsx'
export { default as EventAvailable } from './EventAvailable.jsx'
export { default as EventBusy } from './EventBusy.jsx'
export { default as ExitToApp } from './ExitToApp.jsx'
export { default as ExperienceTicket } from './ExperienceTicket.jsx'
export { default as ExternalLink } from './ExternalLink.jsx'
export { default as Facebook } from './Facebook.jsx'
export { default as Family } from './Family.jsx'
export { default as FavoriteHotel } from './FavoriteHotel.jsx'
export { default as Filter } from './Filter.jsx'
export { default as Fitness } from './Fitness.jsx'
export { default as Flame } from './Flame.jsx'
export { default as FlightCircle } from './FlightCircle.jsx'
export { default as FlightCoverage } from './FlightCoverage.jsx'
export { default as Flights } from './Flights.jsx'
export { default as FreeCancellation } from './FreeCancellation.jsx'
export { default as Fridge } from './Fridge.jsx'
export { default as Gallery } from './Gallery.jsx'
export { default as Gas } from './Gas.jsx'
export { default as Globe } from './Globe.jsx'
export { default as Golf } from './Golf.jsx'
export { default as GolfTee } from './GolfTee.jsx'
export { default as Gps } from './Gps.jsx'
export { default as Graph } from './Graph.jsx'
export { default as Grid } from './Grid.jsx'
export { default as Group } from './Group.jsx'
export { default as GuestScore } from './GuestScore.jsx'
export { default as Guests } from './Guests.jsx'
export { default as Help } from './Help.jsx'
export { default as History } from './History.jsx'
export { default as Home } from './Home.jsx'
export { default as HotelCircle } from './HotelCircle.jsx'
export { default as Hotels } from './Hotels.jsx'
export { default as House } from './House.jsx'
export { default as Hybrid } from './Hybrid.jsx'
export { default as Icon } from './Icon.jsx'
export { default as Image } from './Image.jsx'
export { default as ImageWide } from './ImageWide.jsx'
export { default as Inclusive } from './Inclusive.jsx'
export { default as Information } from './Information.jsx'
export { default as InformationOutline } from './InformationOutline.jsx'
export { default as Instagram } from './Instagram.jsx'
export { default as Key } from './Key.jsx'
export { default as KingBed } from './KingBed.jsx'
export { default as Kitchen } from './Kitchen.jsx'
export { default as Kitchenette } from './Kitchenette.jsx'
export { default as Laptop } from './Laptop.jsx'
export { default as LateNight } from './LateNight.jsx'
export { default as License } from './License.jsx'
export { default as List } from './List.jsx'
export { default as ListContained } from './ListContained.jsx'
export { default as LocalBar } from './LocalBar.jsx'
export { default as Lock } from './Lock.jsx'
export { default as Logout } from './Logout.jsx'
export { default as Loyalty } from './Loyalty.jsx'
export { default as Luggage } from './Luggage.jsx'
export { default as LuggageNo } from './LuggageNo.jsx'
export { default as Manual } from './Manual.jsx'
export { default as Map } from './Map.jsx'
export { default as Mask } from './Mask.jsx'
export { default as Menu } from './Menu.jsx'
export { default as Microwave } from './Microwave.jsx'
export { default as Mileage } from './Mileage.jsx'
export { default as Minus } from './Minus.jsx'
export { default as MoreHorizontal } from './MoreHorizontal.jsx'
export { default as MoreVertical } from './MoreVertical.jsx'
export { default as MultiAirline } from './MultiAirline.jsx'
export { default as MultiOccupancy } from './MultiOccupancy.jsx'
export { default as Museum } from './Museum.jsx'
export { default as Navigation } from './Navigation.jsx'
export { default as NewRelease } from './NewRelease.jsx'
export { default as NoSeat } from './NoSeat.jsx'
export { default as Notification } from './Notification.jsx'
export { default as NotificationImportant } from './NotificationImportant.jsx'
export { default as Notifications } from './Notifications.jsx'
export { default as NotificationsAdd } from './NotificationsAdd.jsx'
export { default as OfflineBolt } from './OfflineBolt.jsx'
export { default as OnDemand } from './OnDemand.jsx'
export { default as Overnight } from './Overnight.jsx'
export { default as Palette } from './Palette.jsx'
export { default as ParentChild } from './ParentChild.jsx'
export { default as ParentChildCircle } from './ParentChildCircle.jsx'
export { default as Parking } from './Parking.jsx'
export { default as PersonalItem } from './PersonalItem.jsx'
export { default as Pets } from './Pets.jsx'
export { default as Phone } from './Phone.jsx'
export { default as PhoneActive } from './PhoneActive.jsx'
export { default as PhoneAgent } from './PhoneAgent.jsx'
export { default as PhoneLog } from './PhoneLog.jsx'
export { default as Picture } from './Picture.jsx'
export { default as PieChart } from './PieChart.jsx'
export { default as Pin } from './Pin.jsx'
export { default as PinAdd } from './PinAdd.jsx'
export { default as PinEdit } from './PinEdit.jsx'
export { default as Plus } from './Plus.jsx'
export { default as PlusOne } from './PlusOne.jsx'
export { default as Pool } from './Pool.jsx'
export { default as Power } from './Power.jsx'
export { default as Priceline } from './Priceline.jsx'
export { default as Printer } from './Printer.jsx'
export { default as Quilt } from './Quilt.jsx'
export { default as RadioChecked } from './RadioChecked.jsx'
export { default as RadioEmpty } from './RadioEmpty.jsx'
export { default as RadioMinus } from './RadioMinus.jsx'
export { default as RadioPlus } from './RadioPlus.jsx'
export { default as Railway } from './Railway.jsx'
export { default as Receipt } from './Receipt.jsx'
export { default as Refresh } from './Refresh.jsx'
export { default as Restaurant } from './Restaurant.jsx'
export { default as Ribbon } from './Ribbon.jsx'
export { default as RibbonAlt } from './RibbonAlt.jsx'
export { default as Road } from './Road.jsx'
export { default as RoomSize } from './RoomSize.jsx'
export { default as Rowing } from './Rowing.jsx'
export { default as SaveAlt } from './SaveAlt.jsx'
export { default as Search } from './Search.jsx'
export { default as SearchRecent } from './SearchRecent.jsx'
export { default as Seat } from './Seat.jsx'
export { default as SeatBusiness } from './SeatBusiness.jsx'
export { default as SeatEconomy } from './SeatEconomy.jsx'
export { default as Security } from './Security.jsx'
export { default as Share } from './Share.jsx'
export { default as Shuttle } from './Shuttle.jsx'
export { default as Sick } from './Sick.jsx'
export { default as SingleBed } from './SingleBed.jsx'
export { default as SingleOccupancy } from './SingleOccupancy.jsx'
export { default as Sites } from './Sites.jsx'
export { default as Sliders } from './Sliders.jsx'
export { default as Smartphone } from './Smartphone.jsx'
export { default as Smoking } from './Smoking.jsx'
export { default as SmokingAllowed } from './SmokingAllowed.jsx'
export { default as SoapBar } from './SoapBar.jsx'
export { default as Sort } from './Sort.jsx'
export { default as Spa } from './Spa.jsx'
export { default as Sparkle } from './Sparkle.jsx'
export { default as SplitTicket } from './SplitTicket.jsx'
export { default as SprayBottle } from './SprayBottle.jsx'
export { default as Star } from './Star.jsx'
export { default as StarHalf } from './StarHalf.jsx'
export { default as SteeringWheel } from './SteeringWheel.jsx'
export { default as Store } from './Store.jsx'
export { default as Subway } from './Subway.jsx'
export { default as SubwayAlt } from './SubwayAlt.jsx'
export { default as Success } from './Success.jsx'
export { default as SuccessOutline } from './SuccessOutline.jsx'
export { default as Sunrise } from './Sunrise.jsx'
export { default as Sustainability } from './Sustainability.jsx'
export { default as Swap } from './Swap.jsx'
export { default as SwapHorizontal } from './SwapHorizontal.jsx'
export { default as Sync } from './Sync.jsx'
export { default as TableChart } from './TableChart.jsx'
export { default as Task } from './Task.jsx'
export { default as TaskChecked } from './TaskChecked.jsx'
export { default as Taxi } from './Taxi.jsx'
export { default as Tea } from './Tea.jsx'
export { default as ThumbsDown } from './ThumbsDown.jsx'
export { default as ThumbsUp } from './ThumbsUp.jsx'
export { default as Timer } from './Timer.jsx'
export { default as Traffic } from './Traffic.jsx'
export { default as Train } from './Train.jsx'
export { default as Trash } from './Trash.jsx'
export { default as TravelExplore } from './TravelExplore.jsx'
export { default as TrendingUp } from './TrendingUp.jsx'
export { default as Trophy } from './Trophy.jsx'
export { default as Tune } from './Tune.jsx'
export { default as Twitter } from './Twitter.jsx'
export { default as Unlock } from './Unlock.jsx'
export { default as User } from './User.jsx'
export { default as UserAdd } from './UserAdd.jsx'
export { default as UserChecked } from './UserChecked.jsx'
export { default as UserCircle } from './UserCircle.jsx'
export { default as Verified } from './Verified.jsx'
export { default as ViewList } from './ViewList.jsx'
export { default as ViewModule } from './ViewModule.jsx'
export { default as Visibility } from './Visibility.jsx'
export { default as VisibilityOff } from './VisibilityOff.jsx'
export { default as Walk } from './Walk.jsx'
export { default as Warning } from './Warning.jsx'
export { default as WarningOutline } from './WarningOutline.jsx'
export { default as WashingHands } from './WashingHands.jsx'
export { default as WeatherHurricane } from './WeatherHurricane.jsx'
export { default as WeatherLightning } from './WeatherLightning.jsx'
export { default as WeatherRain } from './WeatherRain.jsx'
export { default as WeatherSnow } from './WeatherSnow.jsx'
export { default as WeatherSnowRain } from './WeatherSnowRain.jsx'
export { default as WeatherWindy } from './WeatherWindy.jsx'
export { default as Web } from './Web.jsx'
export { default as Whirlpool } from './Whirlpool.jsx'
export { default as Widgets } from './Widgets.jsx'
export { default as Wifi } from './Wifi.jsx'
export { default as Youtube } from './Youtube.jsx'
export { default as ZoomOut } from './ZoomOut.jsx'