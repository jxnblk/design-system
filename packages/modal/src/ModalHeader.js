import React from 'react'
import styled from 'styled-components'
import PropTypes from 'prop-types'
import { Flex, CloseButton, deprecatedPropType, Text } from 'pcln-design-system'

const HeaderWrapper = styled(Flex)`
  height: 40px;
`

const Title = styled(Text)`
  display: flex;
  align-items: center;
`

const StyledCloseButton = styled(CloseButton)`
  height: 24px;
  width: 24px;
  svg {
    vertical-align: top;
  }

  &:focus {
    background-color: ${(props) => props.theme.palette.primary.dark};
    outline: none;
  }
`

const ModalHeader = ({ bg, color, onClose, title }) => (
  <HeaderWrapper align='center' alignItems='center' color={color} bg={bg} px={3}>
    {title && (
      <Title fontSize={1} bold>
        {title}
      </Title>
    )}
    {onClose && <StyledCloseButton onClick={onClose} ml='auto' />}
  </HeaderWrapper>
)

ModalHeader.displayName = ModalHeader

ModalHeader.propTypes = {
  bg: deprecatedPropType('color'),
  onClose: PropTypes.func,
  color: PropTypes.string,
  title: PropTypes.string,
}

ModalHeader.defaultProps = {
  bg: 'blue',
  color: 'white',
}

export default ModalHeader
