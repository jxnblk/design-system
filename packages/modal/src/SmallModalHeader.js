import { Box, deprecatedPropType } from 'pcln-design-system'
import PropTypes from 'prop-types'
import React from 'react'
import styled from 'styled-components'

const HeaderBar = styled(Box)`
  height: 8px;
`

const SmallModalHeader = ({ bg, color }) => <HeaderBar bg={bg} color={color} />

SmallModalHeader.displayName = SmallModalHeader

SmallModalHeader.propTypes = {
  bg: deprecatedPropType('color'),
  color: PropTypes.string,
}

SmallModalHeader.defaultProps = {
  bg: 'green',
  color: 'secondary.base',
}

export default SmallModalHeader
