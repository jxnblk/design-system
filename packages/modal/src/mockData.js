export const currencies = [
  {
    code: 'USD',
    label: 'U.S. Dollar',
    symbol: '$',
  },
  {
    code: 'AUS',
    label: 'Austrailian Dollar',
    symbol: 'AU$',
  },
  {
    code: 'BZR',
    label: 'Brazilian Real',
    symbol: 'R$',
  },
  {
    code: 'CAD',
    label: 'Canadian Dollar',
    symbol: 'C$',
  },
  {
    code: 'GBP',
    label: 'Pound Sterling',
    symbol: '£',
  },
  {
    code: 'CNY',
    label: 'Chinese Yuan Renminbi',
    symbol: '¥',
  },
  {
    code: 'DKK',
    label: 'Danish Krone',
    symbol: 'kr',
  },
  {
    code: 'AED',
    label: 'Emirati Dirham',
    symbol: 'AED',
  },
  {
    code: 'EUR',
    label: 'Euro',
    symbol: '€',
  },
  {
    code: 'HKD',
    label: 'Hong Kong Dollar',
    symbol: 'HK$',
  },
]
