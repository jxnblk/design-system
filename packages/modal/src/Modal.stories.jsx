import { action } from '@priceline/storybook-preset'
import { Box, Button, Text } from 'pcln-design-system'
import { Menu, MenuItem } from 'pcln-menu'
import PropTypes from 'prop-types'
import React from 'react'
import styled from 'styled-components'
import { Modal, ModalHeader, ScrollLock, SmallModalHeader } from '../src'
import { currencies } from './mockData'

const StyledModal = styled(Modal)`
  height: ${(props) => props.height};
`
const CUSTOM_ANIMATION = (transitionState) => `
  transform: translateY(-122%);
  transition: transform 0.3s linear;
  ${transitionState === 'entered' ? `transform: translateY(0%);` : ''}
`

class ModalStory extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      isOpen: props.isOpen,
    }
    this.scrollLock = new ScrollLock()
  }

  static propTypes = {
    isOpen: PropTypes.bool,
    lock: PropTypes.bool,
  }

  render() {
    return (
      <div style={{ height: '1500px' }}>
        <Button
          onClick={() => {
            if (this.props.lock) {
              this.scrollLock.on()
            }
            this.setState({ isOpen: true })
          }}
        >
          Open
        </Button>
        <StyledModal
          ariaLabel='Storybook modal.'
          isOpen={this.state.isOpen}
          onClose={() => {
            this.setState({ isOpen: false })
            if (this.props.lock) {
              this.scrollLock.off()
            }
          }}
          {...this.props}
        >
          <div style={{ height: '1000px' }}>Content with 1000px height</div>
        </StyledModal>
      </div>
    )
  }
}

export default {
  title: 'pcln-modal/Modal',
  component: Modal,
}

export const Raw = () => <ModalStory width={['100px', '200px', '500px']} disableCloseButton />

export const WithSmallModalHeader = () => (
  <ModalStory header={<SmallModalHeader />} width={['80vw', '400px', '500px']} />
)

WithSmallModalHeader.story = {
  name: 'With SmallModalHeader',
}

export const WithModalHeader = () => {
  // Generate content to demonstrate a scrollable <body>
  const contentLines = [...Array(100).keys()]

  return (
    <div>
      <h1>Scroll down to open modal</h1>
      {contentLines.map((i, idx) => (
        <div key={idx}>Line {idx}</div>
      ))}
      <ModalStory
        header={<ModalHeader title='Modal title' onClose={action('Modal closed!')} />}
        height={['90vh', '460px', '560px']}
        width={['80vw', '400px', '500px']}
      />
    </div>
  )
}

export const WithSmallModalHeaderAndWhiteBackgroundModalHeaderWithSetTextStyle = () => {
  return (
    <ModalStory
      header={
        <>
          <SmallModalHeader bg={null} color='primary.base' />
          <ModalHeader
            bg={null}
            color='background.lightest'
            onClose={action('Modal closed!')}
            pt={2}
            textStyle='heading4'
            title='Modal title'
          />
        </>
      }
      height={['90vh', '460px', '560px']}
      width={['80vw', '400px', '500px']}
    />
  )
}

export const WithOverflow = () => (
  <ModalStory header={<SmallModalHeader />} width={['80vw', '400px', '500px']} enableOverflow />
)

export const WithOverflowAndDropdown = () => (
  <StyledModal ariaLabel='Storybook modal.' isOpen={true} header={<SmallModalHeader />} height='100px'>
    <Menu idx='currency-selector' buttonNode={<div>open</div>} size='twoColumns'>
      {currencies.map((currency, index) => {
        const selected = 'USD' === currency.code
        return (
          <MenuItem key={index} selected={selected}>
            <Text regular width={32} textAlign='center' mr={3}>
              {currency.symbol}
            </Text>
            <Text regular>{currency.label}</Text>
          </MenuItem>
        )
      })}
    </Menu>
  </StyledModal>
)

export const WithOverflowAndTallContent = () => (
  <StyledModal
    ariaLabel='Storybook modal.'
    isOpen={true}
    header={<ModalHeader onClose={action('Modal closed!')} />}
    height='100px'
    enableOverflow
    disableCloseButton
  >
    <Box color='primary' height='500px'>
      Priceline
    </Box>
  </StyledModal>
)

export const WithImagemodeAndColorful = () => (
  <ModalStory
    bg='orange'
    header={<SmallModalHeader />}
    width={['100px', '200px', '500px']}
    imgMode
    disableCloseButton
  />
)

export const RoundedModalWithHeader = () => (
  <ModalStory
    bg='orange'
    header={<ModalHeader title='Priceline' />}
    width={['100px', '200px', '500px']}
    imgMode
    disableCloseButton
  />
)

WithImagemodeAndColorful.story = {
  name: 'With imagemode and colorful',
}

export const WithCustomAnimation = () => (
  <ModalStory
    header={<SmallModalHeader />}
    width={['100px', '200px', '500px']}
    dialogAnimation={CUSTOM_ANIMATION}
    verticalAlignment='top'
  />
)

WithCustomAnimation.story = {
  name: 'With custom animation',
}

export const ZeroTimeout = () => (
  <ModalStory disableCloseButton width={['100px', '200px', '500px']} timeout={0} />
)
