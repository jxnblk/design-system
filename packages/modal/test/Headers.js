import React from 'react'
import { render } from '@testing-library/react'
import { ModalHeader, SmallModalHeader } from '../src/index'
import { ThemeProvider, theme } from 'pcln-design-system'

const customRender = (node, ...options) => {
  const rendered = render(
    <ThemeProvider theme={theme}>{node}</ThemeProvider>,
    options
  )

  return {
    ...rendered,
    rerender: (ui, options) =>
      customRender(ui, { container: rendered.container, ...options }),
  }
}

describe('SmallModalHeader', () => {
  const { rerender } = customRender(<SmallModalHeader bg='secondary' />)

  test('render', () => {
    const { container } = rerender(<SmallModalHeader bg='secondary' />)
    expect(container.firstChild).toMatchSnapshot()
  })
})

describe('ModalHeader', () => {
  const { rerender } = customRender(<ModalHeader bg='secondary' />)

  test('render', () => {
    const { container } = rerender(
      <ModalHeader
        bg='secondary'
        title='test'
        onClose={() => {
          console.log('close')
        }}
      />
    )
    expect(container.firstChild).toMatchSnapshot()
  })
})
