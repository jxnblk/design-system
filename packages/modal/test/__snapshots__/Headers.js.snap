// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalHeader render 1`] = `
.c9 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
}

.c8 {
  outline: none;
}

.c3 {
  padding-left: 16px;
  padding-right: 16px;
  background-color: #0a0;
  color: #fff;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c4 {
  font-size: 14px;
  font-weight: 700;
}

.c7 {
  -webkit-font-smoothing: antialiased;
  display: inline-block;
  vertical-align: middle;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: inherit;
  font-weight: 600;
  line-height: 1.5;
  cursor: pointer;
  border-radius: 2px;
  border-width: 0;
  border-style: solid;
  font-size: 14px;
  padding: 9.5px 18px;
  margin-left: auto;
  background-color: #007aff;
  color: #fff;
}

.c7:hover {
  background-color: #049;
  color: #fff;
}

.c6 {
  padding: 0;
  background-color: transparent;
  color: inherit;
}

.c6:hover {
  background-color: transparent;
  color: inherit;
}

.c6 > div {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c0 {
  font-family: 'Montserrat','Helvetica Neue',Helvetica,Arial,sans-serif;
  line-height: 1.4;
  font-weight: 500;
}

.c0 * {
  box-sizing: border-box;
}

.c1 {
  height: 40px;
}

.c5 svg {
  vertical-align: top;
}

.c5:focus {
  background-color: #049;
  outline: none;
}

<div
  class="c0"
>
  <div
    class="c1 c2 c3"
    color="white"
  >
    <div
      class="c4"
      font-size="1"
    >
      test
    </div>
    <button
      class="c5 c6 c7"
      color="primary"
      title="close"
    >
      <div>
        <svg
          aria-hidden="true"
          class="c8 c9"
          fill="currentcolor"
          focusable="false"
          height="24"
          role="img"
          tabindex="-1"
          viewBox="0 0 24 24"
          width="24"
        >
          <path
            d="M19 6.4L17.6 5 12 10.6 6.4 5 5 6.4l5.6 5.6L5 17.6 6.4 19l5.6-5.6 5.6 5.6 1.4-1.4-5.6-5.6L19 6.4z"
          />
        </svg>
      </div>
    </button>
  </div>
</div>
`;

exports[`SmallModalHeader render 1`] = `
.c2 {
  background-color: #0a0;
}

.c0 {
  font-family: 'Montserrat','Helvetica Neue',Helvetica,Arial,sans-serif;
  line-height: 1.4;
  font-weight: 500;
}

.c0 * {
  box-sizing: border-box;
}

.c1 {
  height: 8px;
}

<div
  class="c0"
>
  <div
    class="c1 c2"
  />
</div>
`;
